uarray/_uarray_dispatch.cxx

```diff
@@ -220,17 +481,17 @@ PyObject * register_backend(PyObject * /* self */, PyObject * args) {
   if (domain.empty())
     return nullptr;
 
-  global_domain_map[domain].registered.push_back(py_ref::ref(backend));
+  global_state[domain].registered.push_back(py_ref::ref(backend));
   Py_RETURN_NONE;
 }
 
 void clear_single(const std::string & domain, bool registered, bool global) {
-  auto domain_globals = global_domain_map.find(domain);
-  if (domain_globals == global_domain_map.end())
+  auto domain_globals = global_state.find(domain);
+  if (domain_globals == global_state.end())
```

Comment:
You should be using `use_local... ? local : global` everywhere, right. For that reason, it would probably be cleaner to just have a pointer to the appropriate map.